aws/src/main/java/org/apache/iceberg/aws/s3/S3OutputStream.java

```diff
@@ -389,7 +391,9 @@ private void completeUploads() {
       S3RequestUtil.configureEncryption(awsProperties, requestBuilder);
       S3RequestUtil.configurePermission(awsProperties, requestBuilder);
 
-      s3.putObject(requestBuilder.build(), RequestBody.fromInputStream(contentStream, contentLength));
+      s3.putObject(
+          requestBuilder.build(),
+          RequestBody.fromContentProvider(contentProvider, contentLength, Mimetype.MIMETYPE_OCTET_STREAM));
```

Comment:
Could this use `RequestBody.fromFile(f)` instead, like the `uploadParts` method does?

Reply:
The input stream here is a composition from multiple `stagingFiles`, is size of `stagingFiles` always 1 when it's not multipart? It's possible if that's the case, but I think the current way is a bit more elegant and future proof without that assumption.

Reply:
Ah, you're right. We're concatenating them. Sounds good!